modules/gateways/btcpay/createinvoice.php

```diff
@@ -119,6 +120,10 @@
     die('[ERROR] In modules/gateways/btcpay/createinvoice.php: Invoice error: ' . var_export($invoice['error'], true));
 } else {
     if (!empty($invoice['data']['url'])) {
+        $is_tor_enabled = preg_match("/\.onion$/", $_SERVER['HTTP_HOST']) && $options['btcpayUrlTor'] != '';
+        if ($is_tor_enabled) {
+            $invoice['data']['url'] = str_replace($options['btcpayUrl'], $options['btcpayUrlTor'], $invoice['data']['url']);
```

Comment:
this could cause a problem if you have btcpayUrl entered with trailing slash and btcpayUrlTor without traling slash so the url ends up like bla.onioni/INVOICEID instead of bla.onion/i/INVOICEID; not sure if there is something done by WHMCS to ensure the url always has or has not a trailing slash

Reply:
@ndeet what do you think about this patch?

```patch
--- a/modules/gateways/btcpay/createinvoice.php
+++ b/modules/gateways/btcpay/createinvoice.php
@@ -110,8 +110,8 @@ $options['redirectURL'] = !empty($GATEWAY['redirectURL']) ? $GATEWAY['redirectUR
 $options['apiKey'] = $GATEWAY['apiKey'];
 $options['transactionSpeed'] = $GATEWAY['transactionSpeed'];
 $options['currency'] = $currency;
-$options['btcpayUrl'] = $GATEWAY['btcpayUrl'];
-$options['btcpayUrlTor'] = $GATEWAY['btcpayUrlTor'];
+$options['btcpayUrl'] = strpos($GATEWAY['btcpayUrl'], "/") ? $GATEWAY['btcpayUrl'] : $GATEWAY['btcpayUrlTor'] . "/";
+$options['btcpayUrlTor'] = strpos($GATEWAY['btcpayUrlTor'], "/") ? $GATEWAY['btcpayUrlTor'] : $GATEWAY['btcpayUrlTor'] . "/";
```

Reply:
It is a bit tricky to detect that way depindig if the URL is entered with protocol or not, more bulletproof is to do a `rtrim($value, "/")` which makes sure there is no trailing slash at the end and then add it again when needed.

Reply:
I've commited your solution